### ISSUE TITLE
[FEATURE] Support TableSchema Catalog to manage table schema (schema registry)

### ISSUE THREAD

**Issue by coolderli** [apache/gravitino#5230]
### Describe the feature

In Kafka and Fileset, we may need a table schema to deserialize data. We can  manage the external schema registry in Gravitino. 

### Motivation

- Manage the table schema in Gravitino. 
- Using SQL to read the Kafka topic or Fileset when binding the table schema to the topic or fileset.

### Describe the solution

We can introduce a TableSchemaCatalog to manage the TableSchema. 

We can bind a table schema such as `catalog.schema.table-schema` to a topic or fileset when needed. So we can get the table schema from the external schema registry. We can also add a schema registry managed by gravitino, so we can directory save the table schema to the gravitino metastore.

![img_v3_02fu_4b0b7b27-c424-4a5b-a0fc-42c6fb54318l](https://github.com/user-attachments/assets/31950e21-464d-4795-bfef-18a52eb6b188)

**Comment by coolderli** [comment]
@jerryshao @shaofengshi @caican00 @xloya @lw-yang What do you think? Any other thoughts about this?

**Comment by xloya** [comment]
I think it is a good idea to manage Schema as a resource at the same level as Table/Fileset. In this way, we can distinguish between Managed Schema (data type is based on Gravitino) and External Schema (data type is based on the existing external Schema Registry). Then, in resources that require a specific Schema (such as some Filesets), we can bind a Schema to it. When obtaining Fileset metadata, we will also obtain the corresponding Schema and use it in some clients.

**Comment by jerryshao** [comment]
It's a bit strange that "schema" is an entity. Theoretically, the entity maps a data object, whereas "schema" is binding to the entity.

**Comment by coolderli** [comment]
@jerryshao Yes. But we can manage the 'schema' in gravitino for better binding. We can locate the schema with `catalog.schema.table-schema`, it's easier to bind it to a data object.
